main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -437,6 +483,50 @@ object BuildServerProtocol {
     }
   }
 
+  private def sbtBuildTargetTask: Def.Initialize[Task[List[BuildTarget]]] = Def.taskDyn {
+    val structure = buildStructure.value
+    val loadedUnits = structure.units.values.toSeq.distinct
+
+    val scalaProvider = appConfiguration.value.provider().scalaProvider()
+    appConfiguration.value.provider().mainClasspath()
+    val scalaJars = scalaProvider.jars()
+    val compileData = ScalaBuildTarget(
+      scalaOrganization = ScalaArtifacts.Organization,
+      scalaVersion = scalaProvider.version(),
+      scalaBinaryVersion = binaryScalaVersion(scalaProvider.version()),
+      platform = ScalaPlatform.JVM,
+      jars = scalaJars.toVector.map(_.toURI.toString)
+    )
+    val workspace = bspWorkspaceSetting.value
+    Def.task {
```

Comment:
This task does not seem to be needed, in which case the `sbtBuildTargetTask` does not need to be dynamic.